src/rebar3_sbom_cyclonedx.erl

```diff
@@ -1,22 +1,42 @@
 -module(rebar3_sbom_cyclonedx).
 
--export([bom/1, bom/2, uuid/0]).
+-export([bom/3, bom/4, uuid/0]).
 
-bom(Components) ->
-    bom(Components, uuid()).
+-define(APP, "rebar3_sbom").
+-define(DEFAULT_VERSION, "1").
+-define(COMPONENT_FIELDS, [name, version, author, description, licenses, purl, sha256]).
 
-bom(Components, Serial) ->
-    Bom = {bom, [{serialNumber, Serial}, {xmlns, "http://cyclonedx.org/schema/bom/1.1"}], [
-        {components, [], [component(Component) || Component <- Components, Component /= undefined]}
-    ]},
+-include_lib("xmerl/include/xmerl.hrl").
+
+bom(File, Components, Opts) ->
+    bom(File, Components, Opts, uuid()).
+
+bom(File, Components, Opts, Serial) ->
+    ValidComponents = lists:filter(fun(E) -> E =/= undefined end, Components),
+    Content = {bom, [{version, ?DEFAULT_VERSION},
+                     {serialNumber, Serial},
+                     {xmlns, "http://cyclonedx.org/schema/bom/1.4"}],
+               [{metadata, metadata()},
+                {components, [], [component(Component) || Component <- ValidComponents]},
+                {dependencies, [], [dependency(Component) || Component <- ValidComponents]}]},
+    Normalized = xmerl_lib:normalize_element(Content),
+    Bom = update_version(File, Normalized, Opts),
     xmerl:export_simple([Bom], xmerl_xml).
 
+metadata() ->
+    [{timestamp, [calendar:system_time_to_rfc3339(erlang:system_time(second))]},
+     {tools, [{tool, [?APP]}]}
```

Comment:
I tried importing into DependencyTrack, but it chokes on this because each 'tool' node should have name, author, version, etc. child nodes.

```suggestion
     {tools, [{tool, [{name, ?APP}]}]}
```